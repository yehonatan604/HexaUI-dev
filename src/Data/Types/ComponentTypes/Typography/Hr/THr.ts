import { TComponent, ComponentOptions } from "../../TComponent";

export type THr = TComponent<"hr"> & {
    options?: HrOptions;
}

export type HrOptions = ComponentOptions & {
    color?: string;
    width?: string;
    height?: string;
}
