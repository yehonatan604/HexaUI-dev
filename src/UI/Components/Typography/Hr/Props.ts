import { TComponent } from "../../../../Data/Types/TComponent";

export type HrProps = TComponent<"hr"> & {
    borderColor?: string;
    borderWidth?: string;
    borderHeight?: string;
};