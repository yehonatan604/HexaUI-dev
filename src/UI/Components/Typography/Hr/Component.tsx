import useColorPalette from "../../../..//Core/Hooks/useColorPallete";
import { THr } from "../../../../Data/Types/ComponentTypes/Typography/Hr/THr";

const Hr = (props: THr) => {
  const { options, className, ...componenetProps } = props;
  const { palette } = useColorPalette();

  const lineWidth = options?.width || "1/2";
  const lineColor = options?.color || palette.standrad;

  return (
    <hr
      className={`border-1 border-${lineColor} w-${lineWidth} m-auto my-2 ${className}`}
      {...componenetProps}
    />
  );
};

export default Hr;
