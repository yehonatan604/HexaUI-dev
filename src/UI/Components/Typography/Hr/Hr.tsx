import { THr } from "./types/THr";

const Hr = (props: THr) => {
  const { options, className, ...componenetProps } = props;

  const lineWidth = options?.width || "1/2";
  const lineColor = options?.color || "standard";

  return (
    <hr
      className={`border-1 border-${lineColor} w-${lineWidth} m-auto my-2 ${className}`}
      {...componenetProps}
    />
  );
};

export default Hr;
