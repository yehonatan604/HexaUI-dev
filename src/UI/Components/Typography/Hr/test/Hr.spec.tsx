import { render, screen } from "@testing-library/react";
import Hr from "../Hr"; // Adjust the import path
import { describe, it, expect } from "vitest";

describe("Hr Component", () => {
  it("should render the Hr component", () => {
    render(<Hr options={{}} />);

    // Check if the hr element is rendered
    const hrElement = screen.getByRole("separator");
    expect(hrElement).toBeInTheDocument();
  });

  it("should apply custom color and width", () => {
    render(
      <Hr
        options={{
          color: "blue-500",
          width: "3/4",
        }}
      />
    );

    const hrElement = screen.getByRole("separator");

    // Check if the custom color and width are applied
    expect(hrElement).toHaveClass("border-blue-500", "w-3/4");
  });

  it("should apply additional class names", () => {
    render(<Hr options={{}} className="custom-class" />);

    const hrElement = screen.getByRole("separator");

    // Check if the additional class name is applied
    expect(hrElement).toHaveClass("custom-class");
  });

  it("should render and match the snapshot", () => {
    const { asFragment } = render(
      <Hr
        options={{
          color: "red-500",
          width: "full",
        }}
        className="custom-class"
      />
    );

    // Snapshot the Hr structure
    expect(asFragment()).toMatchSnapshot();
  });
});
