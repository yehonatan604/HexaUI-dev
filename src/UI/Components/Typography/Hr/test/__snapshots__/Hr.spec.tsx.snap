// Vitest Snapshot v1

exports[`Hr Component > should render and match the snapshot 1`] = `
<DocumentFragment>
  <hr
    class="border-1 border-red-500 w-full m-auto my-2 custom-class"
  />
</DocumentFragment>
`;
